[COMP] Typography and `<Text>` improvements

andrewrubin opened the pull request:
## Description
This PR adds the following:
- The base rem size used in the `calc-rem` utility is now a CSS custom property on the root: `--wtc-base-rem-size`. I've removed the ability to pass it as an argument to the helper.
- Two new font-family CSS custom properties are now available, and set on the `Text` component: `--wtc-font-family-heading` and `--wtc-font-family-body`.
- The `<Text>` component now allows for breakpoint-specific horizontal alignment on the `align` prop (see video).
- Adds a new `wrap` prop which corresponds to the CSS `text-wrap` property; defaults to "pretty".

## Does this close any open issues?
Closes #153 
Closes #162 
Closes #180 

## Notes
- This PR depends on others which are currently open, so please only concern yourself with changes inside the `components/text` directory.

## Screenshots
Here's the breakpoint alignment in action:
https://github.com/wethegit/component-library/assets/30575213/96158a5a-2645-4587-9bc5-a281050d238a

Here's the new `wrap` prop
https://github.com/wethegit/component-library/assets/30575213/fcb974d5-df7a-49d4-9658-bab31e0e2200

marlonmarcello reviewed (dismissed):
Awesome! I think in general we should move away from Sass variables wherever we can.

andrewrubin commented:
Added the wrapping option per #180 as well. Here's a screengrab of that in action (the prop is `wrap`, I recorded the video before I renamed it).

In order to avoid an extra CSS class, I default the `.text` class to use `text-wrap: pretty`, as it's likely to be most commonly used. You can override that via `wrap="wrap"`, `wrap="balance"`, `wrap="nowrap"`, etc.


https://github.com/wethegit/component-library/assets/30575213/fcb974d5-df7a-49d4-9658-bab31e0e2200

marlonmarcello reviewed (approved):
Looking really good!
